icu4j/main/core/src/main/java/com/ibm/icu/segmenter/LocalizedSegmenter.java

```diff
@@ -0,0 +1,188 @@
+// © 2025 and later: Unicode, Inc. and others.
+// License & terms of use: https://www.unicode.org/copyright.html
+
+package com.ibm.icu.segmenter;
+
+import com.ibm.icu.text.BreakIterator;
+import com.ibm.icu.util.ULocale;
+import java.util.Locale;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ * Performs segmentation according to the rules defined for the locale.
+ */
+public class LocalizedSegmenter implements Segmenter {
+
+  private ULocale locale;
+
+  private SegmentationType segmentationType;
+
+  @Override
+  public Segments segment(CharSequence s) {
+    return new LocalizedSegments(s, this);
+  }
+
+  /**
+   * @return a builder for constructing {@code LocalizedSegmenter}
+   * @draft ICU 78
+   */
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  LocalizedSegmenter(ULocale locale, SegmentationType segmentationType) {
+    this.locale = locale;
+    this.segmentationType = segmentationType;
+  }
+
+  /**
+   * @internal
+   * @deprecated This API is ICU internal only.
+   */
+  @Override
+  @Deprecated
+  public BreakIterator getNewBreakIterator() {
+    BreakIterator breakIter;
+    switch (this.segmentationType) {
+      case LINE:
+        breakIter = BreakIterator.getLineInstance(this.locale);
+        break;
+      case SENTENCE:
+        breakIter = BreakIterator.getSentenceInstance(this.locale);
+        break;
+      case WORD:
+        breakIter = BreakIterator.getWordInstance(this.locale);
+        break;
+      case GRAPHEME_CLUSTER:
+      default:
+        breakIter = BreakIterator.getCharacterInstance(this.locale);
+        break;
+    }
+    return breakIter;
+  }
+
+  /**
+   * The type of segmentation to be performed. See the ICU User Guide page
+   * <a
+   * href="https://unicode-org.github.io/icu/userguide/boundaryanalysis/#four-types-of-breakiterator">Boundary Analysis</a>
+   * for further details.
+   */
+  public enum SegmentationType {
+    GRAPHEME_CLUSTER,
+    WORD,
+    LINE,
+    SENTENCE,
+  }
+
+  /**
+   * Builder for {@link LocalizedSegmenter}
+   * @draft ICU 78
+   */
+  public static class Builder {
+
+    private ULocale locale = ULocale.ROOT;
+
+    private SegmentationType segmentationType = SegmentationType.GRAPHEME_CLUSTER;
```

Comment:
Did we discuss this default? I don't remember.
While the locale default is reasonably obvious, a default segmentation type does not seem obvious to me.
I think it would be better to init this to null and have the build() method throw an exception if it's not set.

Reply:
Done. We didn't discuss the default. The rationale for a null for segmentation type, and how that differs from locale, makes sense. I added the validation & exception when the default null isn't overwritten.